aten/src/ATen/StorageImpl.cpp

```diff
@@ -15,14 +18,22 @@ THStorage::THStorage(
       allocator(allocator),
       finalizer(nullptr) {}
 
-THStorage::THStorage(
+StorageImpl::StorageImpl(
     at::ScalarType scalar_type,
     ptrdiff_t size,
     at::Allocator* allocator,
     char flag)
-    : THStorage(
+    : StorageImpl(
           scalar_type,
           size,
           allocator->allocate(at::elementSize(scalar_type) * size),
           allocator,
           flag) {}
+
+Type& StorageImpl::type() const {
+  if (data_ptr.device().is_cuda())
```

Comment:
Does this work with variables?  It looks like this is only used in one place anyway, it's probably just better to do exactly what you need there rather than implement this generally.

Reply:
Sounds good.

Reply:
Yeah, I definitely don't want this method on StorageImpl